src/connection.js

```diff
@@ -259,54 +343,79 @@ class Connection extends EventEmitter {
 
         this.config.options.enableQuotedIdentifier = config.options.enableQuotedIdentifier;
       }
+      deprecateNullFallbackToDefaultConfigValue('options.enableQuotedIdentifier', config.options.enableQuotedIdentifier);
 
       if (config.options.encrypt != undefined) {
+        deprecateNonBooleanConfigValue('options.encrypt', config.options.encrypt);
         this.config.options.encrypt = config.options.encrypt;
       }
+      deprecateNullConfigValue('options.encrypt', config.options.encrypt);
 
       if (config.options.fallbackToDefaultDb != undefined) {
+        deprecateNonBooleanConfigValue('options.fallbackToDefaultDb', config.options.fallbackToDefaultDb);
         this.config.options.fallbackToDefaultDb = config.options.fallbackToDefaultDb;
       }
+      deprecateNullConfigValue('options.fallbackToDefaultDb', config.options.fallbackToDefaultDb);
 
       if (config.options.instanceName != undefined) {
+        deprecateNonStringConfigValue('options.instanceName', config.options.instanceName);
+
         this.config.options.instanceName = config.options.instanceName;
         this.config.options.port = undefined;
       }
+      deprecateNullConfigValue('options.instanceName', config.options.instanceName);
 
       if (config.options.isolationLevel) {
         this.config.options.isolationLevel = config.options.isolationLevel;
       }
 
       if (config.options.language != undefined) {
+        deprecateNonStringConfigValue('options.language', config.options.language);
+
         this.config.options.language = config.options.language;
       }
+      deprecateNullFallbackToDefaultConfigValue('options.language', config.options.language);
 
       if (config.options.localAddress != undefined) {
         this.config.options.localAddress = config.options.localAddress;
       }
 
       if (config.options.multiSubnetFailover != undefined) {
+        deprecateNonBooleanConfigValue('options.multiSubnetFailover', config.options.multiSubnetFailover);
+
         this.config.options.multiSubnetFailover = !!config.options.multiSubnetFailover;
       }
 
-      if (config.options.packetSize) {
+      if (config.options.packetSize != undefined) {
+        if (config.options.packetSize <= 0 || config.options.packetSize >= 65536) {
+          throw new RangeError('Port must be > 0 and < 65536');
```

Comment:
Error is thrown for port number should be packetsize 😉 , also this is a breaking change, it's better to include it in breaking changes PR

Reply:
I wouldn't _quite_ classify this as a breaking change (passing any other value here would make the connection fail), but you're right, let's just put it into the breaking changes pull request. 👍